### PR TITLE
Per unit

### PR DESCRIPTION
**Pull request by chargio**
Adding per-unit management in chargeback

### CODE REVIEW

File: app/models/manageiq/consumption/showback_rate.rb [hunk 1]
```diff
@@ -24,51 +25,58 @@ class ShowbackRate < ApplicationRecord
...
+      # TODO event.resource.type should be eq to category
+      a = dimension.split('#')
+      value, measure = event.get_measure(a[0], a[1])
+      rate_with_values(value, measure,event.time_span, event.month_duration)
     end
 
-    def rate_with_values(value, time_span, cycle_duration, date = Time.now)
-      send(calculation.downcase, value || 0, time_span, cycle_duration, date)
+    def rate_with_values(value, measure, time_span, cycle_duration, date = Time.current)
+      send(calculation.downcase, value, measure, time_span, cycle_duration, date)
```
Comment (lpichler) [inline app/models/manageiq/consumption/showback_rate.rb:47]: maybe guard condition against `VALID_RATE_CALCULATIONS` ?
`raise unless VALID_RATE_CALCULATIONS.includes?(calculation)`

Comment (chargio) [reply app/models/manageiq/consumption/showback_rate.rb:47]: I added the guard condition in the model. You can't have an instance that is not part of VALID_RATE_CALCULATIONS

Comment (chargio) [reply app/models/manageiq/consumption/showback_rate.rb:47]: validates :calculation, :presence => true, :inclusion => { :in => VALID_RATE_CALCULATIONS }

**Review by aljesusg — APPROVED** [review]
LGFM

**Pull request merged**